src/profile/profile.jl

```diff
@@ -0,0 +1,56 @@
+struct MixedModelProfile{T<:AbstractFloat}
+    m::LinearMixedModel{T}    # Model that has been profiled
+    tbl::Table                # Table containing ζ, σ, β, and θ from each conditional fit
+    fwd::Dict{Symbol}         # Interpolation splines for ζ as a function of a parameter
+    rev::Dict{Symbol}         # Interpolation splines for a parameter as a function of ζ
+end
+
+include("utilities.jl")
+include("fixefpr.jl")
+include("sigmapr.jl")
+include("thetapr.jl")
+include("vcpr.jl")
+
+"""
+    profile(m::LinearMixedModel; threshold = 4)
+
+Return a `MixedModelProfile` for the objective of `m` with respect to the fixed-effects coefficients.
+
+`m` is `refit!` if `!isfitted(m)`.
+
+Profiling starts at the parameter estimate and continues until reaching a parameter bound or the absolute
+value of ζ exceeds `threshold`.
+"""
+function profile(m::LinearMixedModel{T}; threshold=4) where {T}
```

Comment:
```suggestion
function profile(m::LinearMixedModel; threshold=4)
```

I don't think we use the type parameter anywhere in the method body